modules/passkey/test/userStories/executeUserOpFromPasskeySigner.ts

```diff
@@ -0,0 +1,147 @@
+ * The passkey signer then signs a userOp and gets executed. The test case here assumes that a signer contract is already deployed.
+ *
+ * The flow can be summarized as follows:
+ * Step 1: Setup the contracts.
+ * Step 2: Create a userOp and sign it using passkey credential.
+ * Step 3: Execute the userOp.
+ */
+describe('Execute userOp from Passkey signer [@userstory]', () => {
+  const setupTests = deployments.createFixture(async ({ deployments }) => {
+    const { EntryPoint, Safe4337Module, SafeProxyFactory, SafeModuleSetup, SafeL2, FCLP256Verifier, WebAuthnSignerFactory } =
+      await deployments.run()
+
+    const [relayer] = await ethers.getSigners()
+
+    const entryPoint = await ethers.getContractAt('IEntryPoint', EntryPoint.address)
+    const module = await ethers.getContractAt(Safe4337Module.abi, Safe4337Module.address)
+    const proxyFactory = await ethers.getContractAt(SafeProxyFactory.abi, SafeProxyFactory.address)
+    const safeModuleSetup = await ethers.getContractAt(SafeModuleSetup.abi, SafeModuleSetup.address)
+    const singleton = await ethers.getContractAt(SafeL2.abi, SafeL2.address)
+    const verifier = await ethers.getContractAt('IP256Verifier', FCLP256Verifier.address)
+    const signerFactory = await ethers.getContractAt('WebAuthnSignerFactory', WebAuthnSignerFactory.address)
+
+    const navigator = {
+      credentials: new WebAuthnCredentials(),
+    }
+
+    const credential = navigator.credentials.create({
+      publicKey: {
+        rp: {
+          name: 'Safe',
+          id: 'safe.global',
+        },
+        user: {
+          id: ethers.getBytes(ethers.id('chucknorris')),
+          name: 'chucknorris',
+          displayName: 'Chuck Norris',
+        },
+        challenge: ethers.toBeArray(Date.now()),
+        pubKeyCredParams: [{ type: 'public-key', alg: -7 }],
+      },
+    })
+
+    // Deploy a signer contract
+    const publicKey = decodePublicKey(credential.response)
+    await signerFactory.createSigner(publicKey.x, publicKey.y, await verifier.getAddress())
+    const signer = await signerFactory.getSigner(publicKey.x, publicKey.y, await verifier.getAddress())
+
+    // Deploy a Safe with passkey signer as owner
+
+    // The initializer data to enable the Safe4337Module as a module on a Safe
+    const initializer = safeModuleSetup.interface.encodeFunctionData('enableModules', [[module.target]])
+
+    // Create setup data to deploy a Safe with EOA and passkey signer as owners, threshold 1, Safe4337Module as module and fallback handler
+    const setupData = singleton.interface.encodeFunctionData('setup', [
+      [signer],
+      1n,
+      safeModuleSetup.target,
+      initializer,
+      module.target,
+      ethers.ZeroAddress,
+      0,
+      ethers.ZeroAddress,
+    ])
+
+    // Deploy a Safe with EOA and passkey signer as owners
+    const safeSalt = Date.now()
+    const safeAddress = await proxyFactory.createProxyWithNonce.staticCall(singleton, setupData, safeSalt)
```

Comment:
nit: if something is doable off-chain, it's better to do that off-chain. We should have a utility function for address calculation somewhere

Reply:
In general, I agree, but with the user story I think it is nice to use view methods to simplify the code and cognitive load when reading the code here. Implementing this off-chain would just add noise IMO.

Reply:
Alternatively, safeAddress can be retrieved from event logs as it was done in this test previously

Reply:
Shall we perhaps mention in a comment that an alternative off-chain address calculation via create2 is available?

It's not a dealbreaker for me, but IMO, if we write an example for something, we can expect this example to be followed